packages/modules-sdk/src/types/index.ts

```diff
@@ -1 +1,101 @@
-export * from "./module"
+import { Logger as _Logger } from "winston"
+import { MedusaContainer as coreMedusaContainer } from "medusa-core-utils"
+
+export type MedusaContainer = coreMedusaContainer
+export type Constructor<T> = new (...args: any[]) => T
+
+export type LogLevel =
+  | "query"
+  | "schema"
+  | "error"
+  | "warn"
+  | "info"
+  | "log"
+  | "migration"
+export type LoggerOptions = boolean | "all" | LogLevel[]
+
+export type Logger = _Logger & {
+  progress: (activityId: string, msg: string) => void
+  info: (msg: string) => void
+  warn: (msg: string) => void
+}
+
+export enum MODULE_SCOPE {
+  INTERNAL = "internal",
+  EXTERNAL = "external",
+}
+
+export enum MODULE_RESOURCE_TYPE {
+  SHARED = "shared",
+  ISOLATED = "isolated",
+}
+
+export type InternalModuleDeclaration = {
+  scope: MODULE_SCOPE.INTERNAL
+  resources: MODULE_RESOURCE_TYPE
+  dependencies?: string[]
+  resolve?: string
+  options?: Record<string, unknown>
+}
+
+export type ExternalModuleDeclaration = {
+  scope: MODULE_SCOPE.EXTERNAL
+  server: {
+    type: "http"
+    url: string
+    keepAlive: boolean
+  }
+}
+
+export type ModuleResolution = {
+  resolutionPath: string | false
+  definition: ModuleDefinition
+  options?: Record<string, unknown>
+  dependencies?: string[]
+  moduleDeclaration?: InternalModuleDeclaration | ExternalModuleDeclaration
+}
+
+export type ModuleDefinition = {
+  key: string
+  registrationName: string
+  defaultPackage: string | false
+  label: string
+  canOverride?: boolean
+  isRequired?: boolean
+  dependencies?: string[]
+  defaultModuleDeclaration:
+    | InternalModuleDeclaration
+    | ExternalModuleDeclaration
+}
+
+export type LoaderOptions = {
```

Comment:
suggestion: `<TOptions = Record<string, unknown>` and then `options?: TOptions` That way when using it (like in the inventory) you can directly type the options and you don't need to cast everywhere anymore. If you accept that you should also change in the places it is used to remove the force cast

Reply:
I will apply that but the casts will change to a properly typed definition on each module in a next PR.